viper.go

```diff
@@ -115,6 +115,22 @@ func (fnfe ConfigFileNotFoundError) Error() string {
 	return fmt.Sprintf("Config File %q Not Found in %q", fnfe.name, fnfe.locations)
 }
 
+// ConfigFileAlreadyExistsError denotes failure to write new configuration file.
+type ConfigFileAlreadyExistsError string
+
+// Error returns the formatted error when configuration already exists.
+func (faee ConfigFileAlreadyExistsError) Error() string {
+	return fmt.Sprintf("Config File %q Already Exists", string(faee))
+}
+
+// MissingConfigurationError denotes a required configuration setting has not been provided.
+type MissingConfigurationError string
```

Comment:
I wonder if it makes sense to have a separate error type for this. Not found and already exists kinda make sense to me, but this is hardly a recoverable error, so I would probably replace the usage of this error with `errors.New`.

WDYT?

Reply:
Seems reasonable to me!  I'll update that.